main/core.js

```diff
@@ -119,36 +129,36 @@ async function start (ctx) {
     .pipe(split2())
     .on('data', line => logs.pushLine(line))
 
-  /** @type {string | null} */
-  let exitReason = null
-
-  app.on('before-quit', () => {
-    childProcess.kill()
-  })
-
-  childProcess.on('close', code => {
-    console.log(`Core closed all stdio with code ${code ?? '<no code>'}`)
-
-    if (code === 2) {
-      // FIL_WALLET_ADDRESS did not pass our screening. There is not much
-      // we can do about that, there is no point in reporting this error
-      // to Sentry.
-      return
-    }
-
-    Sentry.captureException('Core exited', scope => {
-      // Sentry UI can't show the full 100 lines
-      scope.setExtra('logs', logs.getLastLines(10))
-      scope.setExtra('reason', exitReason)
-      return scope
-    })
-  })
+  const onBeforeQuit = () => childProcess.kill()
+  app.on('before-quit', onBeforeQuit)
+
+  const onceExited = once(childProcess, 'exit')
+  const onceClosed = once(childProcess, 'close')
+
+  const [exitCode, exitSignal] = await onceExited
+  app.removeListener('before-quit', onBeforeQuit)
+  const reason = exitSignal
+    ? `via signal ${exitSignal}`
+    : `with code: ${exitCode}`
+  const msg = `Core exited ${reason}`
+  console.log(msg)
+  const exitReason = exitSignal || exitCode ? reason : null
+
+  const [closeCode] = await onceClosed
+  console.log(`Core closed all stdio with code ${closeCode ?? '<no code>'}`)
+
+  if (closeCode === 2) {
+    // FIL_WALLET_ADDRESS did not pass our screening. There is not much
+    // we can do about that, there is no point in reporting this error
+    // to Sentry.
+    throw new Error('Don\'t restart')
```

Comment:
What happens when this error is thrown? Is this going to crash the entire Station Desktop app? I guess that's acceptable, considering that this condition should never happen.

However, I think it will be confusing when a user is presented with "uncaught error" window saying "don't restart".

Let's rework this error message to better explain to the reader what happened.

Reply:
Like this?